src/order/category/LinearOrder.lean

```diff
@@ -21,11 +27,38 @@ attribute [derive [large_category, concrete_category]] LinearOrder
 
 instance : has_coe_to_sort LinearOrder Type* := bundled.has_coe_to_sort
 
-/-- Construct a bundled LinearOrder from the underlying type and typeclass. -/
+/-- Construct a bundled `LinearOrder` from the underlying type and typeclass. -/
 def of (α : Type*) [linear_order α] : LinearOrder := bundled.of α
 
 instance : inhabited LinearOrder := ⟨of punit⟩
 
 instance (α : LinearOrder) : linear_order α := α.str
 
+instance has_forget_to_PartialOrder : has_forget₂ LinearOrder PartialOrder :=
+bundled_hom.forget₂ _ _
+
+/-- Constructs an equivalence between linear orders from an order isomorphism between them. -/
+@[simps] def iso_of_order_iso {α β : LinearOrder.{u}} (e : α ≃o β) : α ≅ β :=
+{ hom := e,
+  inv := e.symm,
+  hom_inv_id' := by { ext, exact e.symm_apply_apply x },
+  inv_hom_id' := by { ext, exact e.apply_symm_apply x } }
+
+/-- `order_dual` as a functor. -/
+@[simps] def to_dual : LinearOrder ⥤ LinearOrder :=
+{ obj := λ X, of (order_dual X), map := λ X Y, order_hom.dual }
+
+/-- The equivalence between `PartialOrder` and itself induced by `order_dual` both ways. -/
+def dual_equiv : LinearOrder ≌ LinearOrder :=
```

Comment:
```suggestion
/-- The equivalence between `PartialOrder` and itself induced by `order_dual` both ways. -/
@[simps]
def dual_equiv : LinearOrder ≌ LinearOrder :=
```
then you can remove `dual_equiv_functor` and `dual_equiv_inverse` below

Reply:
Nope! For some reason, it creates too many simp lemmas, which end up not being in simp normal form because of each other. It looks like it's trying some `bundled` projections.

Reply:
`@[simps functor inverse]` should work instead then